presto-main/src/main/java/com/facebook/presto/type/DecimalSaturatedFloorCasts.java

```diff
@@ -135,6 +137,35 @@ public static Slice doubleToLongDecimal(double value, int resultPrecision, int r
         return encodeUnscaledValue(bigDecimalToBigintFloorSaturatedCast(BigDecimal.valueOf(value), resultPrecision, resultScale));
     }
 
+    public static final SqlScalarFunction FLOAT_TO_DECIMAL_SATURATED_FLOOR_CAST = SqlScalarFunction.builder(DecimalSaturatedFloorCasts.class)
+            .signature(Signature.builder()
+                    .kind(SCALAR)
+                    .operatorType(SATURATED_FLOOR_CAST)
+                    .argumentTypes(FLOAT.getTypeSignature())
+                    .returnType(parseTypeSignature("decimal(result_precision,result_scale)", ImmutableSet.of("result_precision", "result_scale")))
+                    .build()
+            )
+            .implementation(b -> b
+                    .methods("floatToShortDecimal", "floatToLongDecimal")
+                    .withExtraParameters((context) -> {
+                        int resultPrecision = Ints.checkedCast(context.getLiteral("result_precision"));
+                        int resultScale = Ints.checkedCast(context.getLiteral("result_scale"));
+                        return ImmutableList.of(resultPrecision, resultScale);
+                    })
+            ).build();
+
+    @UsedByGeneratedCode
+    public static long floatToShortDecimal(long value, int resultPrecision, int resultScale)
+    {
+        return bigDecimalToBigintFloorSaturatedCast(BigDecimal.valueOf(intBitsToFloat((int) value)), resultPrecision, resultScale).longValueExact();
```

Comment:
make it consistent with `doubleToShortDecimal`, `doubleToLongDecimal` floor cast as discussed.
Floor casting to `int` seems to be too aggressive (e.g. when table data is all fractional)